lib/gds_api/publishing_api_v2.rb

```diff
@@ -43,4 +56,14 @@ def links_url(content_id)
   def publish_url(content_id)
     "#{endpoint}/v2/content/#{content_id}/publish"
   end
+
+  def merge_optional_keys(params, options, optional_keys)
+    optional_keys.inject(params) do |hash, optional_key|
```

Comment:
This could be simpler using `each_with_object`. You wouldn't need the `else` clause that way.